### DISCUSSION TITLE
Vyxal SE Chat Bot ideas

### DISCUSSION THREAD

**Discussion by AMiller42** [Vyxal/Vyxal#152] [Ideas]
**Now that the bot is in a public repo, [any new ideas should be submitted there instead](https://github.com/Vyxal/VyxalBotSE/issues).**

**Comment by AMiller42** [comment]
Command idea: @VyxalBot lol u good?

Check bot's current status, make sure everything's functioning properly.

**Reply by hyperneutrino** [reply]
This has been implemented with the syntax `@VyxalBot` (case-insensitive) followed by `(status|((lol )?u good( (there )?(my )?(bro|dude|sis|buddy|mate|m8)?)?\??))`.

**Comment by mathcat4** [comment]
Command idea: @Vyxalbot execute [code] [inputs] [flags]
Executes the vyxal code and sends a new message as the output

**Reply by ysthakur** [reply]
Might want to switch up the order and use some non-Vyxal character like `⍝` to separate the three sections.

**Reply by mathcat4** [reply]
@ysthakur you mean [inputs]⍝[flags]⍝[code]?

**Reply by hyperneutrino** [reply]
This has been implemented with the syntax `@VyxalBot` (case-insensitive) followed by `(exec(ute)?|run|run code|eval(uate)?)` followed by the code in a code block, optionally flags in a code block, and then as many inputs in code blocks as you want. Multiline code is not supported and will require a different syntax.

**Comment by mathcat4** [comment]
Command idea: @Vyxalbot wat Vyxal iz?
Sends links to the interpreter, github repo, the org, the tuto, the codepage and the elements.md

**Reply by AMiller42** [reply]
I think this is a very good idea, especially if someone new stops by with questions about Vyxal. However, I think maybe we should change the wording of the command to be in the spirit of Vyxal, such as:

@VyxalBot wat Vyxal iz?

**Reply by mathcat4** [reply]
@AMiller42 Maybe @Vyxalbot What's Vyxal? A bug?

Just joking

**Reply by hyperneutrino** [reply]
This has been implemented with the syntax `@VyxalBot` (case-insensitive) followed by `(h[ea]lp|wh?at( i[sz]|'s)? vyxal|what vyxal i[sz])\?*$`.

**Comment by ysthakur** [comment]
Command idea: @VyxalBot Please run this Vyxal code - Puts in a link to the interpreter, but the code is just lyxal, so you get rickrolled.

**Reply by AMiller42** [reply]
@VyxalBot plz give Lyxal link - It's stating exactly what it will do, namely give a link to `lyxal`, but it's subtle enough that nobody will notice, or if they do they will just think it's a typo.

**Reply by ysthakur** [reply]
That's probably a better idea.

**Reply by mathcat4** [reply]
good idea

**Reply by hyperneutrino** [reply]
This has been implemented with the syntax `@VyxalBot` (case-insensitve) followed by `(please|pl[sz]) (run this vyxal code|gi(ve|b) lyxal link)$`.

**Comment by ysthakur** [comment]
Feature idea: Notify of new forks (without pinging)

**Reply by hyperneutrino** [reply]
This is already a feature.

**Comment by chunkybanana** [comment]
Feature idea: have a Tamagotchi?

**Reply by ysthakur** [reply]
I fear this will only turn Vyxal Bot into more of a meme bot, but it's probably too late already.

**Reply by mathcat4** [reply]
what's a tamagotchi?

**Reply by hyperneutrino** [reply]
I am declining this feature request because it will require too much work and has nothing to do with Vyxal, and the bot is already bad enough when it comes to noise. If we want experience enhancement features that allow for more fun things, please consult with @Lyxal and other people in the Vyxal room for how we feel about introducing so much noise. I would be willing to set up the chat room to feel more like a friendly lounge than a productive environment but currently it is just development plus memes, and having an environmental/ambient feature would change that significantly.

**Comment by AMiller42** [comment]
We should put together some sort of welcome message for people who have never participated in chat. It could have things like info about Vyxal or the bot, or whatever.
